music_assistant/helpers/audio.py

```
@@ -955,19 +995,11 @@ async def get_multi_file_stream(
     Arguments:
     seek_position: The position to seek to in seconds
     """
-    files_list: list[str] = []
     if not isinstance(streamdetails.path, list):
         raise InvalidDataError("Multi-file streamdetails requires a list of MultiPartPath")
-    skipped_duration = 0.0
-    for part in streamdetails.path:
-        if not isinstance(part, MultiPartPath):
-            raise InvalidDataError("Multi-file streamdetails requires a list of MultiPartPath")
-        if seek_position and part.duration and (skipped_duration + part.duration) < seek_position:
-            skipped_duration += part.duration
-            continue
-        files_list.append(part.path)
-    if seek_position:
-        seek_position -= int(skipped_duration)
+    parts, seek_position = _get_part_from_position(streamdetails.path, seek_position)
+    assert parts is not None
```

I would ditch this assert and just raise an exception in the _get_part_from_position if a invalid combination of params is given